magento-research/pwa-studio
# MiniCart Refactor

## Description

This PR refactors the entirety of the mini cart / right drawer.

Most notably it:
1. Updates all components in `MiniCart/` to functions from classes
1. Adds new components instead of private getters that returned JSX to simplify, increase modularity, and decrease duplication
1. Renames `openOptionsDrawer` and `closeOptionsDrawer` actions to `beginEditItem` and `endEditItem` respectively
1. Fixes a bug where you couldn't un-favorite an item
1. Adds unit tests for all components (and snapshots where appropriate)

## Related Issue




Closes #978 .

## Verification Steps

The mini cart has been completely re-written so please test it thoroughly. At a minimum:

1. Add an item to the cart. Ensure it appears in the cart with the correct options.
1. Favorite and then un-favorite an item in the cart. Ensure the kebab updates correctly.
1. Edit an item in the cart. Ensure it updates correctly.
1. Verify that you can remove an item from the cart.
1. Complete the checkout process.

There are also a lot of new snapshots generated, please do take a look at them for accuracy.

## How Have YOU Tested this?




1. `yarn test`
1. Verification steps above
1. Verify new snapshots for accuracy
1. Run `yarn storybook:venia` and click through the `MiniCart` stories

## Screenshots / Screen Captures (if appropriate):

The mini cart / right drawer:

![Screen Shot 2019-05-22 at 4 13 16 PM](https://user-images.githubusercontent.com/13182778/58209734-41678300-7cad-11e9-9ae9-bd5aef0d49b7.png)

## Proposed Labels for Change Type/Package




This has to be a `major` because I renamed two props that the `MiniCart` accepts:
`openOptionsDrawer` -> `beginEditItem`
`closeOptionsDrawer` -> `endEditItem`

If we're wary of a `major` these changes can be removed easily but the new names are much more clear about what's actually happening.

## Checklist:



## TODO:

- [x] Refactor the `kebab`, `mask`, `product`, and `section` components
- [x] Add Storybook stories for the UI states of `MiniCart`
- [x] Kebab not working?
- [x] Last item not visible?
- [x] Continue Shopping not responsive?
